Assignment 2.Rmd

```diff
@@ -95,6 +102,35 @@ Now practice with data from our class. Please create a **person-network** with t
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
-### To Submit Your Assignment
+```{r}
+library(dplyr)
+library(tidyr)
+hudk4050_classes <- read_csv("hudk4050-classes.csv")
+h1 <- hudk4050_classes%>% unite(Name, `First Name`, `Last Name`, sep= " ")
+h2 <- gather(h1,key="class number", value="class",2:7)
+h2 <- na.omit(h2)
+h2 = select(h2, -c("class number"))
+h2$class <- gsub("[[:space:]]", "", h2$class)
+h2 <- h2%>%filter(Name!='ZIMO CHEN')
+h2 <- h2%>%filter(class!="HUDK4050")
+h2$count=1
+h3 <- spread(h2,class,count)
+h3[is.na(h3)] <- 0
+#h3 is the person-class matrix
+h4 <- h3[,2:59]
+h4 <- t(t(h4)) %*% t(h4)
+colnames(h4) <- h3$Name
+rownames(h4) <- h3$Name
+diag(h4)<- NA
+#h4 is the person-person matrix
+h <- graph_from_adjacency_matrix(h4, mode="directed")
+plot(h,layout=layout.fruchterman.reingold,vertex.size=2,
+     vertex.label.dist=1, vertex.color="red", edge.arrow.size=0.3)
```

Comment:
Nice work 👍  It is difficult to see the graph underneath labels though. Perhaps you could shrink the labels using `vertex.label.cex`